### DISCUSSION TITLE
swaggerClient generates erroneous code

### DISCUSSION THREAD

**Discussion by 11sas11** [amberelectric/public-api#81] [Q&A]
![image](https://user-images.githubusercontent.com/48985019/138627678-d101200d-ac48-49e0-8d97-d5bd0322750c.png)

I get this error when I generate the code adding the reference in Visual Studio.  Any ideas?

**Comment by madpilot** [comment]
Shouldn't startDate and endDate be date objects rather than just generic objects? I don't have any experience with the C# swagger generator, but given that type signature, I would imagine C# doesn't have enough information to work out that that is really a date object.

**Reply by madpilot** [reply]
Straight up casting them, may fix it

**Reply by 11sas11** [reply]
okay, I am pretty amateur at this stuff.  It was autogenerated code, I normally just use the default .ToString() in my code.  Really struggling with it tbh.  I just add the external reference and it won't compile.  For now I just commented it out and it compiles.

**Reply by madpilot** [reply]
That generator isn't very smart. Something like this might do it:

```csharp
PriceAsync(string stringId, DateTime startDate, DateTime endDate, double? resolution)
{

}
```

Might do it

**Reply by 11sas11** [reply]
okay thanks will try it out

**Comment by 11sas11** [comment]
I ended up not using Swagger and just writing my own classes.

**Reply by madpilot** [reply]
Honestly, the API is so simple, rolling by hand is totally a viable solution.
